### ISSUE TITLE
Consider using `gitlint` in CI to check commit messages

### ISSUE THREAD

**Issue by rlouf** [aesara-devs/aesara#1359]
https://github.com/jorisroovers/gitlint should be configured to make sure commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

**Comment by mohitsaxenaknoldus** [comment]
Hey, I'd like to work on this. You can assign it to me.

**Comment by brandonwillard** [comment]
@rlouf, do we still want to do this via `pre-commit`?  I like that approach for its ability to catch formatting issues before developers even create and submit the commits&mdash;as well as the fact that we can run it in CI when people haven't used `pre-commit`.

**Comment by rlouf** [comment]
Yes, we should so this via pre-commit. @mohitsaxenaknoldus you can start with a minimal configuration like this one: https://github.com/blackjax-devs/blackjax/blob/main/.gitlint which should be enough to get started and iterate on.
